test/framework/spec_helpers.go

```diff
@@ -95,8 +100,86 @@ func DumpAllResourcesAndLogs(ctx context.Context, clusterProxy ClusterProxy, art
 	}
 }
 
+type DescribeClusterInput struct {
+	ClusterctlConfigPath string
+	LogFolder            string
+	KubeConfigPath       string
+	Namespace            string
+	Name                 string
+}
+
+func DescribeCluster(ctx context.Context, input DescribeClusterInput) {
+	log.Logf("clusterctl describe cluster %s --show-conditions=all --show-machinesets=true --grouping=false --echo=true --v1beta2", input.Name)
+
+	clusterctlClient, err := clusterctlclient.New(ctx, input.ClusterctlConfigPath)
+	Expect(err).ToNot(HaveOccurred(), "Failed to create the clusterctl client library")
+
+	tree, err := clusterctlClient.DescribeCluster(ctx, clusterctlclient.DescribeClusterOptions{
+		Kubeconfig: clusterctlclient.Kubeconfig{
+			Path:    input.KubeConfigPath,
+			Context: "",
+		},
+		Namespace:               input.Namespace,
+		ClusterName:             input.Name,
+		ShowOtherConditions:     "all",
+		ShowClusterResourceSets: false,
+		ShowTemplates:           false,
+		ShowMachineSets:         true,
+		AddTemplateVirtualNode:  true,
+		Echo:                    true,
+		Grouping:                false,
+		V1Beta2:                 true,
+	})
+	Expect(err).ToNot(HaveOccurred(), "Failed to run clusterctl describe")
+
+	Expect(os.MkdirAll(input.LogFolder, 0750)).To(Succeed(), "Failed to create log folder %s", input.LogFolder)
+
+	f, err := os.Create(filepath.Join(input.LogFolder, fmt.Sprintf("clusterctl-describe-%s.txt", input.Name)))
+	Expect(err).ToNot(HaveOccurred(), "Failed to create a file for saving clusterctl describe output")
+
+	defer f.Close()
+
+	w := bufio.NewWriter(f)
+	cmdtree.PrintObjectTreeV1Beta2(tree, w)
+	Expect(w.Flush()).To(Succeed(), "Failed to save clusterctl describe output")
+}
+
+type DescribeAllClusterInput struct {
+	Client               client.Client
+	KubeConfigPath       string
+	ClusterctlConfigPath string
+	ArtifactFolder       string
+	Namespace            string
+}
+
+func DescribeAllCluster(ctx context.Context, input DescribeAllClusterInput) {
+	clusters := &clusterv1.ClusterList{}
+	Eventually(func() error {
+		return input.Client.List(ctx, clusters, client.InNamespace(input.Namespace))
+	}, retryableOperationTimeout, retryableOperationInterval).Should(Succeed(), "Failed to list clusters in namespace %s", input.Namespace)
+
+	for _, c := range clusters.Items {
+		DescribeCluster(ctx, DescribeClusterInput{
+			ClusterctlConfigPath: input.ClusterctlConfigPath,
+			LogFolder:            filepath.Join(input.ArtifactFolder, "clusters", c.Name),
```

Comment:
I think we should put this file somewhere below

https://gcsweb.k8s.io/gcs/kubernetes-ci-logs/pr-logs/pull/kubernetes-sigs_cluster-api/11966/pull-cluster-api-e2e-blocking-main/1900114995866243072/artifacts/clusters/bootstrap

e.g. here: https://gcsweb.k8s.io/gcs/kubernetes-ci-logs/pr-logs/pull/kubernetes-sigs_cluster-api/11966/pull-cluster-api-e2e-blocking-main/1900114995866243072/artifacts/clusters/bootstrap/resources/quick-start-g2ekd7/Cluster/

The current location https://gcsweb.k8s.io/gcs/kubernetes-ci-logs/pr-logs/pull/kubernetes-sigs_cluster-api/11966/pull-cluster-api-e2e-blocking-main/1900114995866243072/artifacts/clusters/quick-start-km06hh/
is usually for resources from the workload cluster